extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -68,6 +69,57 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     this.storeService_.subscribe(StateProperty.RTL_STATE, (rtlState) => {
       this.onRtlStateUpdate_(rtlState);
     });
+
+    this.storeService_.subscribe(
+      StateProperty.PAGE_SIZE,
+      (pageSizeState) => {
+        this.flipTagIfOffscreen_(pageSizeState);
+      },
+      true /** callToInitialize */
+    );
+  }
+
+  /**
+   * Helper function to check if the shopping tag should flip if it's too far to the right.
+   * @param {!Object} pageSize
+   * @private
+   */
+  flipTagIfOffscreen_(pageSize) {
+    /* We only check the right hand side, as resizing only expands the border to the right. */
+
+    const storyPageWidth = pageSize.width;
+
+    this.measureElement(() => {
+      /*
+       * We are using offsetLeft and offsetWidth instead of getLayoutBox() because
+       * the correct measurements are not taken into account when using a CSS transform (such as translate),
+       * which we are using in the i-amphtml-amp-story-shopping-tag-inner-flipped class.
+       */
+      const {offsetLeft, offsetWidth} = this.element;
+
+      const shouldFlip = offsetLeft + offsetWidth > storyPageWidth;
+
+      this.mutateElement(() => {
+        this.shoppingTagEl_.classList.toggle(
+          'i-amphtml-amp-story-shopping-tag-inner-flipped',
+          shouldFlip
+        );
+
+        const dotEl = this.shoppingTagEl_.querySelector(
+          '.i-amphtml-amp-story-shopping-tag-dot'
+        );
+
+        dotEl.classList.toggle(
+          'i-amphtml-amp-story-shopping-tag-dot-flipped',
+          shouldFlip
+        );
+
+        this.shoppingTagEl_.classList.toggle(
```

Comment:
Since this call is independent of tag flip logic, I think you can move it out of this method and place it right after the `this.flipTagIfOffscreen_()` call in the PAGE_SIZE callback

Reply:
I tired that, and even after adding in a mutate element, it will have problems syncing the flip state during initialization in the browser tests I performed (there is a small window where the tag is not flipped at the beginning, and we are hiding that by setting the opacity to zero in a CSS class). Having this here ensures that the tag will not be shown in that state.